bazelisk.sh

```diff
@@ -12,6 +12,9 @@
 
 set -euo pipefail
 
+# Change to this script's directory, as it is the location of the bazel workspace.
+cd "$(dirname "$0")"
```

Comment:
Should we surround the script with pushd / popd?

Reply:
that would make the behavior where bazelisk.sh build ... suddenly points to everything while bazel build ... points to everything in a directory a bit more confusing in my mind. at least this way there's some indication of what's happening.

Reply:
FYI, generally it's not considered good behavior for shell scripts to change the user's working directory, unless it's specifically an environment changing script that is always run in its own shell (so exiting the shell would bring the user back). Since this PR is explicitly having the script be used as some sort of function call for other projects, I think it'd be best to follow the norms here.

Reply:
Yeah, I was distracted by my other concern.

I'd looked into the pushd popd thing because I had the same concern initially. How are you running this that it changes your working directory unintentionally? The scope of the change is limited to the script's environment unless you do something intentional like ```. bazelisk.sh``` right?

Reply:
That's correct: the script runs in its own environment and does not (cannot) affect the caller's environment.